### PR TITLE
fixes for php8.0

### PR DESCRIPTION
**Pull request by plandolt**
I got an error about php8.0 json package and also forced the installation of specific packages for ast and apcu.

### CODE REVIEW

**Comment by dominikzogg** [comment]
@scuben thanks for the PR, is there no json package yet or won't be, cause its core?

**Comment by plandolt** [comment]
yes i think so. its a virtual package now:

Package: php8.0-json
State: not a real package (virtual)
N: Can't select candidate version from package php8.0-json as it has no candidate
N: Can't select versions from package 'php8.0-json' as it is purely virtual
N: No packages found

File: ansible/roles/php/tasks/main.yml [hunk 1]
```diff
@@ -17,17 +17,21 @@
...
-      - php-apcu
-      - php-ast
+      - "php{{ php.version }}-apcu"
+      - "php{{ php.version }}-ast"
       - php-pear
 
+- name: install php json module for php != 8.0
+  apt:
+    name: "php{{ php.version }}-json"
+  when: php.version|string != '8.0'
```
Comment (dominikzogg) [inline ansible/roles/php/tasks/main.yml]: this solution will work until 8.1 gets released, is there no way  `< 8` ?

Comment (dominikzogg) [reply ansible/roles/php/tasks/main.yml]: `when: php.version < 8` for example

Comment (dominikzogg) [reply ansible/roles/php/tasks/main.yml]: https://github.com/scuben/vagrant-php-ubuntu/pull/1

Comment (dominikzogg) [reply ansible/roles/php/tasks/main.yml]: works with php 7.4, does it with 8.0?

**Comment by dominikzogg** [comment]
@scuben thx

File: ansible/roles/php/tasks/main.yml (hunk 1 above)
Comment (plandolt) [reply ansible/roles/php/tasks/main.yml]: tested now for 7.4 and 8.0. the version can be defined as string or float. only catch is that 8 (int or string) wont work (it needs to be 8.0). I guess same has applied for 7 vs. 7.4 back in the days. branch is therefore ready for merge.

**Pull request merged**